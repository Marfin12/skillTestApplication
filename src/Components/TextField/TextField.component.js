import * as React from 'react';
import { View, Text } from 'react-native';
import { useSelector } from 'react-redux';

import styles from './TextField.styles';
import config from './TextField.config';

const TextField = props => {
  const getSelect = useSelector(({themeReducer}) => themeReducer);
  const theme = getSelect.theme;
  const {darkThemeColor, defaultThemeColor, style, textStyle, children} = props;

  return (
    <View style={style}>
      <Text
        numberOfLines={props.numberOfLines}
        style={[
          styles.text(theme, darkThemeColor, defaultThemeColor),
          textStyle,
        ]}>
        {children}
      </Text>
    </View>
  );
};

TextField.displayName = config.displayName;
TextField.defaultProps = config.defaultProps;

export default TextField;
