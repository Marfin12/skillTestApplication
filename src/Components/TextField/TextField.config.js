const defaultProps = {
  style: {},
  textStyle: {},
  darkThemeColor: '#FFF',
  defaultThemeColor: '#111',
  numberOfLines: 0,
};

const displayName = 'TextField';

export default {
  defaultProps,
  displayName,
};
