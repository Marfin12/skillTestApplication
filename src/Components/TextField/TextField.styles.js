const styles = {
  text: (darkTheme, darkMode, whiteMode) => {
    return {
      color: darkTheme ? darkMode : whiteMode,
    };
  },
};

export default styles;
