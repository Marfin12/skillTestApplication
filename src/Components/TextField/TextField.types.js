// @flow

import * as React from 'react';
import type {
  ViewStyleProp,
  TextStyleProp,
} from 'react-native/Libraries/StyleSheet/StyleSheet';

export type Props = {|
  style: ViewStyleProp,
  textStyle: TextStyleProp,
  darkThemeColor: string,
  defaultThemeColor: string,
  numberOfLines: number,
  children: React.Node
|};
