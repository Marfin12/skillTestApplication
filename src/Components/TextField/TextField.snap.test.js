import React from 'react';
import renderer from 'react-test-renderer';

import TextField from './TextField.component';
import config from './TextField.config';

jest.mock('react-redux', () => {
  const ActualReactRedux = jest.requireActual('react-redux');
  const lightMode = {
    theme: false,
  };
  const darkMode = {
    theme: true,
  };
  return {
    ...ActualReactRedux,
    useSelector: jest
      .fn()
      .mockImplementationOnce(() => lightMode)
      .mockImplementationOnce(() => lightMode)
      .mockImplementationOnce(() => darkMode)
      .mockImplementationOnce(() => darkMode),
  };
});

describe('Light Mode Text Field Component', () => {
  test('renders text field component with default props', () => {
    const ComponentWrapper = renderer
      .create(<TextField {...config.defaultProps} />)
      .toJSON();
    expect(ComponentWrapper).toMatchSnapshot();
  });
  test('renders text field component with elipsis', () => {
    const ComponentWrapper = renderer
      .create(<TextField {...config.defaultProps} numberOfLines={1} />)
      .toJSON();
    expect(ComponentWrapper).toMatchSnapshot();
  });
});

describe('Dark Mode Text Field Component', () => {
  test('renders text field component with default props', () => {
    const ComponentWrapper = renderer
      .create(<TextField {...config.defaultProps} />)
      .toJSON();
    expect(ComponentWrapper).toMatchSnapshot();
  });
  test('renders text field component with elipsis', () => {
    const ComponentWrapper = renderer
      .create(<TextField {...config.defaultProps} numberOfLines={1} />)
      .toJSON();
    expect(ComponentWrapper).toMatchSnapshot();
  });
});
