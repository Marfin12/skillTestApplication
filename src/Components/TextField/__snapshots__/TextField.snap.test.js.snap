// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dark Mode Text Field Component renders text field component with default props 1`] = `
<View
  style={Object {}}
>
  <Text
    numberOfLines={0}
    style={
      Array [
        Object {
          "color": "#FFF",
        },
        Object {},
      ]
    }
  />
</View>
`;

exports[`Dark Mode Text Field Component renders text field component with elipsis 1`] = `
<View
  style={Object {}}
>
  <Text
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#FFF",
        },
        Object {},
      ]
    }
  />
</View>
`;

exports[`Light Mode Text Field Component renders text field component with default props 1`] = `
<View
  style={Object {}}
>
  <Text
    numberOfLines={0}
    style={
      Array [
        Object {
          "color": "#111",
        },
        Object {},
      ]
    }
  />
</View>
`;

exports[`Light Mode Text Field Component renders text field component with elipsis 1`] = `
<View
  style={Object {}}
>
  <Text
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#111",
        },
        Object {},
      ]
    }
  />
</View>
`;
